src/test/java/com/ace/alfox/MoveActionTests.java

```diff
@@ -0,0 +1,65 @@
+package com.ace.alfox;
+
+import com.ace.alfox.game.MoveAction;
+import com.ace.alfox.game.interfaces.IAction;
+import com.ace.alfox.game.models.Player;
+import com.ace.alfox.lib.ActionResult;
+import com.ace.alfox.lib.Database;
+import com.ace.alfox.lib.Vector2;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.HashMap;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertFalse;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class MoveActionTests {
+    Player player;
+    IAction move;
+    static Database db;
+
+    @BeforeClass
+    public static void createConnection() {
+        db = new Database("memory");
+    }
+
+    @Before
+    public void setUp() {
+        player = new Player();
+        move = new MoveAction(db);
```

Comment:
Do you know if we can autowire this and we just have a test configuration that supplies different instances for tests? For example for prod it provides one DB and for tests it provides an in memory db?

not at all a blocker for this PR, just wanted to ask about it :)

Reply:
I couldn't figure out a way, to my knowledge the configurations are all file based. We can inject settings via the command line but I don't know how to configure that in the gradle commands. I'm open to suggestions. I dislike _magic strings_.